plugins/modules/dynamodb_table.py

```diff
@@ -174,300 +195,737 @@
 
-def wait_until_table_active(module, table, wait_timeout):
-    max_wait_time = time.time() + wait_timeout
-    while (max_wait_time > time.time()) and (table.describe()['Table']['TableStatus'] != 'ACTIVE'):
-        time.sleep(5)
-    if max_wait_time <= time.time():
-        # waiting took too long
-        module.fail_json(msg="timed out waiting for table to exist")
+        return index_map
+    except Exception as e:
+        module.fail_json_aws(e, msg='Decode failure', index_data=index_data)
 
 
-def delete_dynamo_table(connection, module):
-    table_name = module.params.get('name')
+def compatability_results(current_table):
+    if not current_table:
+        return dict()
+
+    throughput = current_table.get('provisioned_throughput', {})
 
-    result = dict(
-        region=module.params.get('region'),
-        table_name=table_name,
+    primary_indexes = _decode_primary_index(current_table)
+
+    hash_key_name = primary_indexes.get('hash_key_name')
+    hash_key_type = primary_indexes.get('hash_key_type')
+    range_key_name = primary_indexes.get('range_key_name')
+    range_key_type = primary_indexes.get('range_key_type')
+
+    indexes = list()
+    global_indexes = current_table.get('_global_index_map', {})
+    local_indexes = current_table.get('_local_index_map', {})
+    for index in global_indexes:
+        idx = dict(global_indexes[index])
+        idx.pop('provisioned_throughput', None)
+        indexes.append(idx)
+    for index in local_indexes:
+        idx = dict(local_indexes[index])
+        idx.pop('provisioned_throughput', None)
+        indexes.append(idx)
+
+    compat_results = dict(
+        hash_key_name=hash_key_name,
+        hash_key_type=hash_key_type,
+        range_key_name=range_key_name,
+        range_key_type=range_key_type,
+        indexes=indexes,
+        read_capacity=throughput.get('read_capacity_units', None),
+        region=module.region,
+        table_name=current_table.get('table_name', None),
+        table_status=current_table.get('table_status', None),
+        tags=current_table.get('tags', {}),
+        write_capacity=throughput.get('write_capacity_units', None),
     )
 
+    return compat_results
+
+
+def get_dynamodb_table():
+    table_name = module.params.get('name')
+    try:
+        table = _describe_table(TableName=table_name)
+    except is_boto3_error_code('ResourceNotFoundException'):
+        return None
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:  # pylint: disable=duplicate-except
+        module.fail_json_aws(e, msg='Failed to describe table')
+
+    table = table['Table']
     try:
-        table = Table(table_name, connection=connection)
+        tags = client.list_tags_of_resource(aws_retry=True, ResourceArn=table['TableArn'])['Tags']
+    except is_boto3_error_code('AccessDeniedException'):
+        module.warn('Permission denied when listing tags')
```

Comment:
Would it be better to fail rather than warn ? other modules seem to fail if it can't get the tags